### PR TITLE
Make color picker accessible

### PR DESCRIPTION
**Pull request by bjarnef**
Currently the color picker isn't accessible, e.g. not possible to navigate using keyboard to open color picker and select a color.

It should probably also toggle value of [aria-expanded](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) on the button element, based on popover is open or not and [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) if not inline.

### CODE REVIEW

File: src/spectrum.js [hunk 1]
```diff
@@ -79,10 +79,10 @@
         return contains(style.backgroundColor, 'rgba') || contains(style.backgroundColor, 'hsla');
     })(),
     replaceInput = [
-        "<div class='sp-replacer'>",
+        "<button type='button' class='sp-replacer'>",
             "<div class='sp-preview'><div class='sp-preview-inner'></div></div>",
             "<div class='sp-dd'>&#9660;</div>",
-        "</div>"
+        "</button>"
```
Comment (seballot) [inline src/spectrum.js:85]: Hi !
I don't think div inside a button is valid html :)

Comment (bjarnef) [reply src/spectrum.js:85]: I don't recall if it is valid according to HTML5 specs, but otherwise `<span>` is .. or the `<div>` can use `role="button"`.

**Comment by seballot** [comment]
Hi @bjarnef !

Thanks for the initiative

Please ping me when your work is ready for a review

**Comment by bjarnef** [comment]
@seballot feel free to have a look.. I haven't checked if the enter key select the palette though, but found some inspiration here https://dev.to/tylerjdev/when-role-button-is-not-enough-dac

If the palette "buttons" use `<button>` element it would work out of the box and without `tabindex` attribute.

I guess the `aria-*` attributes also should change according to the settings.

Key press should also focus the handles in the hue and opacity sliders, but it would need similar events on these containers and arrow up/down and left/right to change slider value using keyboard. They could also use home/end key to go to lowest/highest value.

Sometimes Shift + up/down or left/right multiply the value by e.g. 10 to improve the behaviour using  keyboard like in this color picker https://shoelace.style/components/color-picker while up/down or left/right still is used to fine-tune.

**Comment by fabienwnklr** [comment]
> @seballot feel free to have a look.. I haven't checked if the enter key select the palette though, but found some inspiration here https://dev.to/tylerjdev/when-role-button-is-not-enough-dac
> 
> If the palette "buttons" use `<button>` element it would work out of the box and without `tabindex` attribute.
> 
> I guess the `aria-*` attributes also should change according to the settings.
> 
> Key press should also focus the handles in the hue and opacity sliders, but it would need similar events on these containers and arrow up/down and left/right to change slider value using keyboard. They could also use home/end key to go to lowest/highest value.
> 
> Sometimes Shift + up/down or left/right multiply the value by e.g. 10 to improve the behaviour using keyboard like in this color picker https://shoelace.style/components/color-picker while up/down or left/right still is used to fine-tune.

Hi, i just started a [fork](https://github.com/fabienwnklr/spectrum) i wanna try to maintain it better (with more release and more activity i hope), i understand well your PR you can make a PR into my fork and helping me to upgrade it :)

**Comment by seballot** [comment]
hi @fabienwnklr,

If you want to invest energy on this project, you don't need to fork it, I can give you full access to this repo. Please let me know

**Comment by fabienwnklr** [comment]
i just see last commit from `Apr 3, 2022` that why i guess "fork" is better.. but if you're reactive and open to maintain it, we can do it :)

**Comment by seballot** [comment]
The problem with the fork is that you will need to deploy under a new NPM package name, so most people who already switched to this spectrum2 would need to switch again to your new fork

**Comment by fabienwnklr** [comment]
> The problem with the fork is that you will need to deploy under a new NPM package name, so most people who already switched to this spectrum2 would need to switch again to your new fork

i understand, i don't use it with packages manager so i don't have this issue, but i understand

**Comment by bjarnef** [comment]
Any new version to be published containing these changes?

Furthermore it should also implement keyboard events to move the sliders using up/down or left/right  keys.

Sometimes Shift + up/down or left/right multiply the value by e.g. 10 to improve the behaviour using keyboard like in this color picker https://shoelace.style/components/color-picker

**Comment by seballot** [comment]
Hi, hopefully I tested before publishing, your code was introducing a UI bug, and a syntax error in javascript. Did you even test it?

Same for @fabienwnklr did you test it before meging into master?

I have publish a new version 2.0.10

**Pull request merged**